src/main/java/com/direwolf20/buildinggadgets/common/util/blocks/RegionSnapshot.java

```diff
@@ -161,30 +169,30 @@ private static RegionSnapshot deserialize(NBTTagCompound tag) {
                     serializedData.getCompound(TILE_NBT)));
         }
 
-        return new RegionSnapshot(world, region, blockStates.build(), tileData.build());
+        return new RegionSnapshot(world, positions, blockStates.build(), tileData.build());
     }
 
     /**
      * Create a {@link RegionSnapshot} builder.
      * <p>
      * Predicate for whether record blocks or not can be customized through the builder. If no extra conditions are
-     * needed, use {@link #take(World, Region)} instead.
+     * needed, use {@link #take(World, IPositionPlacementSequence)} instead.
      */
-    public static Builder select(World world, Region region) {
-        return new Builder(world, region);
+    public static Builder select(World world, IPositionPlacementSequence positions) {
```

Comment:
How could I miss that! This should at most use IWorld, IWorldWriter might be possible too!

Reply:
No `IWorldWriter` (if we don't want do some random generic trickery like `<T extends IWorldReader & IWorldWriter>`...) because we use both `getBlockState` and `setBlockState`.

Reply:
But we can use IWorld

Reply:
@hnOsmium0001 can you resolve this asap so I can merge this? I want to start on getting all the tools code clear in 1.14

Reply:
This seems not to be possible due to static initilisers, therefore we can't do anything besides alowing foe IWorld to be pased into here.